---
title: "qPCR of vaginal explant and epithelial cell RNA"
date: "March 11, 2016"
output:
  md_document:
    variant: markdown_github
---

These data are from qPCR experiments done by Erik Layton (Corey/Zhu lab) using the same RNA that was used on the epithelial and explant microarray experiments. 

We chose one HSV2 gene (ICP27, Immediate Early) and used beta actin (ACTB) as a housekeeper for the epithelial cell assay. The assays use a FAM dye label and a non-fluorescent quencher. Samples were assayed in replicate.

For both cells and explant experiments, samples had different concentrations but were standardized by volume:

* The same volume of RNA from each sample was used make cDNA, then equal volumes of the resulting cDNA went into the qPCR reaction. 

##Epithelial Cell Results

Caveats for epithelial cell qPCR data:

* We weren't able to find the RNA from the epithelial cells donor 3

* We excluded the sample that was "3E" on the microarray (HVE 4, 24hr, V186) because we wouldnt be able to compare it to the microarray results (it was excluded because of a possible pipetting error).

* I don't think that Erik did a validation for this experiment to show that the efficiencies of the target and refernce were approximately equal, but I assume that they have done this in their lab for other assays using these genes.


```{r, echo=FALSE,warning=FALSE, message= FALSE}

library(dplyr)
library(stringr)
library(plateR)
library(pander)
library(reshape2)
library(ggplot2)


#this is just the CTs and well positions
#from file "2016-03-07 florian lab cell culture.xlsx"

results<-read.csv("input_data_from_Erik/cells CT and well only.csv")

#Adding in the layout that I made manually from Erik's data 
#from file "2016-03-07 florian lab cell culture.xlsx"

dat<-add_plate(file ="input_data_from_Erik/cells qPCR plate layout.csv",results, "Wells")

#There were some wells where I didn't know what the contents were and they were all undetermined. Confirmed with EL that they are controls. I Put unknown for the Gene and and sample in those wells and will remove them here.

dat<-filter(dat, Sample != "unknown")


#load the file with more annotation data for the results(other than sample name)

load("Epithelial_cell_metadata_output/Cells_RNA_for_qPCR.Rdata")

#merge in with results by the common columns
dat_ann<-merge(dat,Cells_RNA_for_qPCR, by.x = "Sample", by.y = "TubeLabel")



#sanity check : dat_ann still has 68 values


```

These replicates got "Undetermined" results from the qPCR machine, but all were Mock samples and the concentration looks ok.

```{r, echo=FALSE, warnings = FALSE}
undet<-dat_ann %>%
  filter(CT=="Undetermined")%>%
  select(TissueID,Treatment,Time,Gene, newNanodrop)
  
  
pander(undet)
```
Analysis using the Relative (ddCT) method:

* For both genes, calculate the average of the sample replicates.

* dCT = Average ICP27 CT - Average ACTB CT 

* ddCT = Treatment dCT - Average Mock dCT
 
* Amount of Target gene normalized to housekeeper, and relative to Mock sample = 2^-ddCT

This is a plot of the amount of the ICP27 gene normalized to the housekeeper and relative to Mock (log10 scale). Each panel shows data from a different tissue donor. The missing sample on the right panel is the one we left out intentionally. 

```{r, echo=FALSE, warnings = FALSE}
### CALCULATIONS
#Steps for analysis:
#http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf


#More here: https://www.thermofisher.com/us/en/home/life-science/pcr/real-time-pcr/qpcr-education/absolute-vs-relative-quantification-for-qpcr.html

#remove the undetermined

dat_ann<-dat_ann %>%
  filter(CT!="Undetermined")



#First make CT numeric so it stops throwing errors
#If you just do as.numeric, R changes it to the factor level,
#NOT the actual number, so change to character first.

dat_ann$CT<-as.numeric(as.character(dat_ann$CT))



# Average the replicates, using just one rep if the other was undet.
avg_dat_ann<-dat_ann%>%
  arrange(Sample)%>%
  group_by(Gene, TissueID, Treatment, Time, Sample)%>%
  summarise(AvgCT = mean(CT))%>%
  ungroup()
  
#put the genes in their own dfs
ACTBavg<-avg_dat_ann%>%
  filter(Gene =="ACTB")

ICP27avg<-avg_dat_ann%>%
  filter(Gene =="ICP27")%>%
  select(-Gene)
 

#add the ACTB data onto the side of the ICP27 data to prepare for subracting
epiAnalysis<-cbind(ICP27avg, ACTBavg$AvgCT)

names(epiAnalysis)[5:6]<-c("ICP27AvgCT","ACTBAvgCT")

#For analysis, subtract the avg ACTB CT from the avg ICP27 CT

#For ddCT, subtract the avg MOCK dCT from each sample dCT

#For fold change, do 2^-ddCT

#fix the factors for the Time variable



epiAnalysis<-epiAnalysis %>%
  mutate(dCT = ICP27AvgCT-ACTBAvgCT)

#get the mean of the mock dCTs
meanMock<-epiAnalysis%>%
  filter(Treatment=="Mock")%>%
  summarise(mean = mean(dCT))



#calculate the ddCT by subtracting the dCT of the MOCK from the dCT of the other samples

epiAnalysis<-epiAnalysis %>%
  mutate(ddCT = dCT-meanMock$mean)%>%
  mutate(TargetRNA = 2^-ddCT)%>%
  filter(Treatment !="Mock")%>%
  arrange(Sample)

epiAnalysis$Time<-factor(epiAnalysis$Time,levels = c("3","8","24"))

epiAnalysis$TissueID<-factor(epiAnalysis$TissueID, levels = c("1","4"))


### It turns out that "V186" should actually be just called "186" so I'm replacing those names here.

epiAnalysis$Treatment<- str_replace(epiAnalysis$Treatment, "V186", "186")


ggplot(epiAnalysis,(aes(x = Time, y = log10(TargetRNA))))+
  geom_point(aes(color = TissueID),size=4.5)+
  geom_line(aes(group = TissueID,color = TissueID))+
  scale_color_discrete( name = "Tissue ID")+
  labs(y = "Fold relative to Mock \n (log10 scale)", x = "Time (hrs)")+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12, vjust = 0.6),
    strip.text = element_text(size =12))+
facet_grid(~Treatment)
  
### DANGER: I have different results than Erik's plot, looks like the labels are mixed up but I don't know what his plate layout looked like or how he assigned the raw results back to the sample names.I rechecked my manual matching of the samples to their well positions below. 
#Update: I found a mistake in his plotting, labels were matched with the wrong data.
#########################################################

#here is a table of the results

# epiAnalysisShort<-epiAnalysis %>%
#   select(TissueID,Treatment, Time,TargetRNA)%>%
#   arrange(TissueID, Time, Treatment)%>%
#   mutate(TargetRNA = format(TargetRNA, digits = 3))
# 
# 
# pander(epiAnalysisShort)


```

##Conclusions

* Similar patterns between the two donors.

* Concentrations of ICP27 increased over time in both V186 and SD90 infected epithelial cell samples. 

* At 3 and 8 hours, there was less ICP27 RNA detected in the samples treated with SD90 than those treated with V186. 

* In the patient where we had data for both treatments at 24hrs, the amount of ICP27 detected was very similar




##Explant Results

For these samples, Erik did not use a housekeeping gene and we didn't use the Mock samples since the explants were not size-standardized. All explants were exposed to the same amount of virus, but if  an explant used for the Mock treatment was bigger than a "treated" explant, and therefore had more cellular RNA, the relative method of analysis would be inaccurate.


Two samples had undetermined values. Maybe the concentrations were too low? For these, I used the value of the other replicate instead of the average of the two.

```{r, echo=FALSE, warnings = FALSE}
##Now I'm reading in the explant data. 

#reading in a copy of the CT data from the spreadsheet that EL gave me. The file from him is called "2016-03-07 florian lab plate 1 explants.xls" I don't have the plate layout for this either but the CTs are aligned with the annotation data so I am guessing that the rows are sorted appropriately.

  
explantCT<-read.csv("input_data_from_Erik/explant CTs only.csv")
  
rep1<-explantCT%>%
  select(-rep2)

names(rep1)[10]<-"CT"


rep2<-explantCT%>%
  select(-rep1)

names(rep2)[10]<-"CT"

explantAnalysis<-rbind(rep1,rep2)

explantUndet<-explantAnalysis%>%
  filter(CT == "Undetermined")%>%
  select(TissueID,Treatment,Timepoint,newNanodrop)

pander(explantUndet)

```

Method for analysis of explant samples: 

* Average the sample replicates

* dCT = 40 - Average CT (because CT of 40 = zero amplification)

* Fold Change = 2^dCT


Here are plots of fold change for each donor. The "Fold Change" is relative to an assumed Mock avg CT of 40. 

```{r, echo=FALSE, warnings = FALSE}

explantAnalysis<-filter(explantAnalysis,CT!="Undetermined")

explantAnalysis$CT<-as.numeric(as.character(explantAnalysis$CT))




#get the avg of the reps
explantAnalysis<-explantAnalysis%>%
  arrange(Sample.ID)%>%
  group_by(TissueID, Treatment, Timepoint, Sample.ID,Well.Position)%>%
  summarise(AvgCT = mean(CT))%>%
  ungroup()%>%
  arrange(Well.Position)%>%
  mutate(dCT = 40-AvgCT)%>%#40 = "zero" because it is the max CT you can get ie no amplification. Basically we are assuming a mock sample here.
  mutate(FoldChange = 2^dCT)


names(explantAnalysis)[3]<-"Time"

explantAnalysis$Time<-factor(explantAnalysis$Time, levels = c("3","8","24"))

explantAnalysis$TissueID<-as.factor(explantAnalysis$TissueID)

### It turns out that "V186" should actually be just called "186" so I'm replacing those names here.

explantAnalysis$Treatment<-str_replace(explantAnalysis$Treatment, "V186","186")


#plot it like it's hot
#by donor


ggplot(explantAnalysis,(aes(x = Time, y = log10(FoldChange))))+
  geom_point(aes(color = Treatment),size = 3.5 )+
  geom_line(aes(color = Treatment, group = Treatment),size=0.5)+
  scale_color_discrete( name = "Tissue ID")+
  labs(y = "Fold Change relative to CT of 40 \n (log 10 scale)", x = "Time (hrs)")+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12, vjust = 0.6 ),
    strip.text = element_text(size =12))+
facet_wrap(~TissueID)


```


Here is a plot of the same data but with all of the the donors together. Black dots are the mean and error bars represent one standard error from the mean.

```{r, echo=FALSE, warnings = FALSE}

#all together

ggplot(explantAnalysis,(aes(x = Time, y = log10(FoldChange))))+
  geom_point(aes(color = TissueID),size = 3.5 )+
  geom_line(aes(color = TissueID, group = TissueID),size=0.5)+
  scale_color_discrete( name = "Tissue ID")+
  stat_summary(fun.data = mean_se)+
  stat_summary(fun.y = mean,geom = "point", size = 2, shape = 95)+
  labs(y = "Fold Change relative to CT of 40 \n (log 10 scale)", x = "Time (hrs)")+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12, vjust = 0.6 ),
    strip.text = element_text(size =12))+
facet_wrap(~ Treatment) 
```


##Conclusions

Erik's remarks on the data:

  * SD90: "What you need to note is that the starting point is very different for each sample.You can't draw a whole lot of conclusions for this, because while you may have added the same amount of virus to each one, the MOI of each sample differed a lot."
  
  * V186: "You guys were worried that your virus didn't infect. It did. It also grew, I can't say how much without some better controls, I'm confident saying that it did proliferate inside the biopsies." 

```{r, echo=FALSE, warnings = FALSE}

sessionInfo()
