---
title: "qPCR of vaginal explant and epithelial cell RNA"
date: "March 11, 2016"
output:
  md_document:
    variant: markdown_github
---

These data are from qPCR experiments done by Erik Layton(Corey/Zhu lab) using the same RNA that was used on the epithelial and explant microarray experiments. 

We chose one HSV2 gene (ICP27, Immediate Early) and used beta actin (ACTB) as a housekeeper for the epithelial cell assay. The assays use a FAM dye label and a non-fluorescent quencher.

##Epithelial Cell Results

Caveats for epithelial cell qPCR data:

We weren't able to find the RNA from the epithelial cells donor 3

We excluded the sample that was "3E" on the microarray (HVE 4, 24hr, V186) because we wouldnt be able to compare it to the microarray results (it was excluded because of a possible pipetting error).


```{r, echo=FALSE,warning=FALSE, message= FALSE}

library(dplyr)
library(stringr)
library(plateR)
library(pander)
library(reshape2)
library(ggplot2)

#I made this plate layout by 

#this is just the CTs and well positions
#from file "2016-03-07 florian lab cell culture.xlsx"

results<-read.csv("cells CT and well only.csv")

#Adding in the layout that I made manually from Erik's data 
#from file "2016-03-07 florian lab cell culture.xlsx"

dat<-add_plate(file="cells qPCR plate layout.csv",results, "Wells")
#There were some wells where I didn't know what the contents were and they were all undetermined. I think they are controls but will check with EL. I Put unknown for the Gene and and sample in those wells and will remove them here.

dat<-filter(dat, Sample != "unknown")


#load the file will more annotation data for the results( other than sample name)
load("Cells_RNA_for_qPCR.Rdata")

#merge in with results by the common columns
dat_ann<-merge(dat,Cells_RNA_for_qPCR, by.x = "Sample", by.y = "TubeLabel")



#sanity check : dat_ann still has 68 values


```

These replicates got "Undetermined" results from the qPCR machine, but all were Mock samples and the concentration looks ok.

```{r, echo=FALSE, warnings = FALSE}
undet<-dat_ann %>%
  filter(CT=="Undetermined")%>%
  select(TissueID,Treatment,Time,Gene, newNanodrop)
  
  
pander(undet)
```
Analysis:

* For both genes, calculate the average of the sample replicates

* dCT = Average ICP27 CT - Average ACTB CT 

* ddCT = Treatment dCT - Average Mock dCT
 
* Amount of target normalized to housekeeper and relative to Mock = 2^-ddCT

This is a plot of the amount of the  ICP27 gene normalized to the housekeeper and relative to Mock. Each panel shows data from a different tissue donor. The missing sample on the right panel is the one we left out intentionally. 

```{r, echo=FALSE, warnings = FALSE}
### CALCULATIONS
#Steps for analysis:
#http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf

#remove the undetermined

dat_ann<-dat_ann %>%
  filter(CT!="Undetermined")



#First make CT numeric so it stops throwing errors
#If you just do as.numeric, R changes it to the factor level,
#NOT the actual number, so change to character first.

dat_ann$CT<-as.numeric(as.character(dat_ann$CT))



# Average the replicates, using just one rep if the other was undet.
avg_dat_ann<-dat_ann%>%
  arrange(Sample)%>%
  group_by(Gene, TissueID, Treatment, Time, Sample)%>%
  summarise(AvgCT = mean(CT))%>%
  ungroup()
  
#put the genes in their own dfs
ACTBavg<-avg_dat_ann%>%
  filter(Gene =="ACTB")

ICP27avg<-avg_dat_ann%>%
  filter(Gene =="ICP27")%>%
  select(-Gene)
 

#add the ACTB data onto the side of the ICP27 data to prepare for subracting
analysis<-cbind(ICP27avg, ACTBavg$AvgCT)

names(analysis)[5:6]<-c("ICP27AvgCT","ACTBAvgCT")

#For analysis, subtract the avg ACTB CT from the avg ICP27 CT

#For ddCT, subtract the avg MOCK dCT from each sample dCT

#For fold change, do 2^-ddCT

#fix the factors for the Time variable



analysis<-analysis %>%
  mutate(dCT = ICP27AvgCT-ACTBAvgCT)

#get the mean of the mock dCTs
meanMock<-analysis%>%
  filter(Treatment=="Mock")%>%
  summarise(mean = mean(dCT))



#calculate the ddCT by subtracting the dCT of the MOCK from the dCT of the other samples

analysis<-analysis %>%
  mutate(ddCT = dCT-meanMock$mean)%>%
  mutate(Target = 2^-ddCT)%>%
  filter(Treatment !="Mock")%>%
  arrange(Sample)

analysis$Time<-factor(analysis$Time,levels = c("3","8","24"))

analysis$TissueID<-factor(analysis$TissueID, levels = c("1","4"))

ggplot(analysis,(aes(x = Time, y = Target)))+
  geom_point(aes(color=Treatment),size=4.5)+
  geom_line(aes(group = Treatment,color=Treatment))+
  scale_y_log10()+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12),
    strip.text = element_text(size =12))+
facet_wrap(~TissueID)
  
  ##### DANGER: I have different results than Erik's plot, looks like the labels are mixed up but I don't know what his plate layout looked like or how he assigned the raw results back to the sample names.I rechecked my manual matching of the samples to their well positions below. 
#Update: I found a mistake in his plotting, labels were matched with the wrong data.
#########################################################
  
```

##Conclusions

* Similar patterns between the two donors.

* Concentrations of ICP27 increased over time in both V186 and SD90 infected epithelial cell samples. 

* SD90 started out lower (relative to housekeeper and Mock), but both reached approximately the same concentration at 8 and 24hr.




##Explant Results

For these samples , Erik did 2 replicates of each sample but no housekeeping gene and we didn't use the Mock samples since the explants were not size-standardized (so we don't know if more housekeeper/Mock =  more expression, or, if that sample just had a bigger explant).


Two samples had undetermined values. Maybe the concentrations were too low? For these, I used the value of the other replicate instead of the average of the two.

```{r, echo=FALSE, warnings = FALSE}
##Now I'm reading in the explant data. 

#reading in a copy of the CT data from the spreadsheet that EL gave me. The file from him is called "2016-03-07 florian lab plate 1 explants.xls" I don't have the plate layout for this either but the CTs are aligned with the annotation data so I am guessing that the rows are sorted appropriately.

  
explantCT<-read.csv("explant CTs only.csv")
  
rep1<-explantCT%>%
  select(-rep2)

names(rep1)[10]<-"CT"


rep2<-explantCT%>%
  select(-rep1)

names(rep2)[10]<-"CT"

explantAnalysis<-rbind(rep1,rep2)

explantUndet<-explantAnalysis%>%
  filter(CT == "Undetermined")%>%
  select(TissueID, Timepoint,Treatment,newNanodrop)

pander(explantUndet)

```

Method for analysis of explant samples: 

We did not use a housekeeper or the Mock as controls since the explants weren't size standardized.

* Average the sample replicates

* dCT = 40 - Average CT (because CT of 40 = zero amplification)


Here are plots of dCT for each donor and then all together

```{r, echo=FALSE, warnings = FALSE}

explantAnalysis<-filter(explantAnalysis,CT!="Undetermined")

explantAnalysis$CT<-as.numeric(as.character(explantAnalysis$CT))

#get the avg of the reps
avg_dCT_explantAnalysis<-explantAnalysis%>%
  arrange(Sample.ID)%>%
  group_by(TissueID, Treatment, Timepoint, Sample.ID,Well.Position)%>%
  summarise(AvgCT = mean(CT))%>%
  ungroup()%>%
  arrange(Well.Position)%>%
  mutate(dCT = 40-AvgCT) #40 = "zero" because it is the max CT you can get


names(avg_dCT_explantAnalysis)[3]<-"Time"

avg_dCT_explantAnalysis$Time<-factor(avg_dCT_explantAnalysis$Time, levels = c("3","8","24"))

avg_dCT_explantAnalysis$TissueID<-as.factor(avg_dCT_explantAnalysis$TissueID)


#plot it like it's hot
#by donor
ggplot(avg_dCT_explantAnalysis,(aes(x = Time, y = dCT)))+
  geom_point(aes(),size=4.5)+
  geom_line(aes(group = Treatment))+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12),
    strip.text = element_text(size =12))+
facet_grid(TissueID~Treatment)


#all together
ggplot(avg_dCT_explantAnalysis,(aes(x = Time, y = dCT)))+
  geom_point(aes(),size=3, alpha = .4)+
  stat_summary(fun.y = mean, geom="point", size = 5.5)+
theme(axis.text = element_text(size = 12),
      axis.title.x = element_text(size=12), 
    axis.title.y = element_text(size=12),
    strip.text = element_text(size =12))+
  facet_wrap(~Treatment)
```


##Conclusion

* In 6 of 7 donors, concentrations of ICP27 were either the same for both treatments or higher for V186 infected explants.

* Erik said that based on his experience, he would say that SD90 did actually infect the explants.
